### ISSUE TITLE
[Refactoring]: Remove arbitrary length limit of some string functions

### ISSUE THREAD

**Issue by fengttt** [matrixorigin/matrixone#18507]
### Why do you want to refactor this code?

```Markdown
Remove a random length limit of some string functions (repeat etc).

Boost varchar/blob max length to 64MB from 64KB.  64KB is too limited for real world use cases.
```

### Describe the solution you'd like

Boost blob size limit, code cleanups.

**Comment by fengttt** [comment]
@m-schen This is left as an future task to bump MaxVarcharlen from 64K to a bigger number, for example 16M.

**Comment by fengttt** [comment]
Fixed by #20713
